### PR TITLE
DAOS-7260 Upgrade Java to depend on jars compatible with hadoop-3.x

### PR DESCRIPTION
**Pull request by zjf2012**
Currently, hadoop-daos depends on hadoop-3.3.0. It can still work with hadoop-2.7.6. But we got jar conflicts when tested with hadoop-3.2.0. Moving forward, we need to support hadoop-3.x which is also default version of spark-3.1.1 we are going to support. So, this ticket is to upgrade other dependencies of [Java] client and make them compatible with hadoop-3.x.

Some adjustment for snapshot release.

Signed-off-by: jiafu zhang <jiafu.zhang@intel.com>

### CODE REVIEW

File: doc/user/spark.md [hunk 1]
```diff
@@ -14,8 +14,8 @@ to access DAOS from Hadoop and Spark.
 
 1. Linux OS
 2. Java 8
-3. Hadoop 2.7.6
-4. Spark 3
+3. Hadoop 3.2 or later
+4. Spark 3.3 or later
```
Comment (carsonwang) [inline doc/user/spark.md:18]: 3.3? Can this still work with Spark 3.0?

Comment (zjf2012) [reply doc/user/spark.md:18]: it's typo. I meant spark 3.1. And spark-3.1.1 starts using hadoop 3.2 as default hadoop version. I am going to test with spark 3.1.1 and hadoop 3.2.0. So what's your suggestion?

Comment (zjf2012) [reply doc/user/spark.md]: change it to 3.0

**Comment by zjf2012** [comment]
developer info in pom.xml is needed. otherwise, you cannot publish to maven repo.

**Comment by zjf2012** [comment]
@johannlombardi please help review and merge this PR. And another ticket, https://jira.hpdd.intel.com/browse/DAOS-7669, is on the way.

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**